### ISSUE TITLE
[Web Component]: Alert Box - design.va.gov accessibility documentation must be updated for the new component usage

### ISSUE THREAD

**Issue by 1Copenut** [department-of-veterans-affairs/va.gov-team#23790]
## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

## Point of Contact

**VFS Point of Contact:** _Trevor_

## Details

The design.va.gov [accessibility documentation for the Alertbox component](https://design.va.gov/components/alertboxes) should be updated to give accurate guidance to teams wanting to use the new component correctly. Specifically I'd like to have the following points updated:

* Advise teams if the `visible` prop is set to true, the alert will correctly render only a div with `aria-live="polite"`. The alert should be rendered inside this div so it's announced properly to screen readers.
* Alert boxes that are always on the page should not be used for emergency information that must be read aloud to screen readers immediately.
* Alert boxes should not have a `role="alertdialog"`. This is not reliably announced to screen readers on first page load and these alert boxes do not rise to the level of  a dialog. They're inline messaging, where a dialog is a captive experience that must be acknowledged or dismissed before users can return to their previous flow.

## Acceptance Criteria

- [ ] Documentation for accessibility is updated
- [ ] Documentation PR is approved by VSP accessibility and content specialists

**Comment by CourtneyChambers77** [comment]
Description
This technique demonstrates how to use the alert role and describes the effect it has on browsers and assistive technology.

The alert role is used to communicate an important and usually time-sensitive message to the user. When this role is added to an element, the browser will send out an accessible alert event to assistive technology products which can then notify the user about it. The alert role is most useful for information that requires the user's immediate attention, for example:

An invalid value was entered into a form field 
The user's login session is about to expire
The connection to the server was lost, local changes will not be saved
Because of its intrusive nature, the alert role must be used sparingly and only in situations where the user's immediate attention is required. Dynamic changes that are less urgent should use a less aggressive method, such as aria-live="polite" or other live region roles.
When the alert role is added to an element, or such an element becomes visible, the user agent should do the following:

Expose the element as having an alert role in the operating system's accessibility API.
Fire an accessible alert event using the operating system's accessibility API if it supports it.
Assistive technology products should listen for such an event and notify the user accordingly:

Screen readers may interrupt current output (whether it's speech or braille) and immediately announce or display the alert message.
Screen magnifiers may visually indicate that an alert occurred and what the alert text was. 

***
The snippet below shows how the alert role is added directly into the html source code. The moment the element finishes loading the screen reader should be notified of the alert. If the element was already in the original source code when the page loaded, the screen reader will announce the error immediately after announcing the page title.
<h2 role="alert">Your form could not be submitted because of 3 validation errors.</h2>
let myAlert = document.createElement("p");
myAlert.setAttribute("role", "alert");
let myAlertText = document.createTextNode("You must agree with our terms of service to create an account.");
myAlert.appendChild(myAlertText);
document.body.appendChild(myAlert);
$("<p role='alert'>You must agree with our terms of service to create an account.</p>").appendTo(document.body);
// When the user tries to submit the form with less than 3 checkboxes selected:
document.getElementById("formInstruction").setAttribute("role", "alert");
// When the user tries to submit the form with less than 3 checkboxes selected:
document.getElementById("formInstruction").setAttribute("role", "alert");
.hidden {
  display:none;
}

<p id="expirationWarning" role="alert" class="hidden">Your log in session will expire in 2 minutes</p>

// removing the 'hidden' class makes the element visible, which will make the screen reader announce the alert:
document.getElementById("expirationWarning").className = "";
